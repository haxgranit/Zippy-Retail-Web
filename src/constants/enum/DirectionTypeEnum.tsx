// eslint-disable-next-line import/prefer-default-export
export enum DirectionTypeEnum {
  OUTBOUND = 'Outbound',
  INBOUND = 'Inbound',
}
