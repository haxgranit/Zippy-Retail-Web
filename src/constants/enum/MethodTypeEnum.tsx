// eslint-disable-next-line import/prefer-default-export
export enum MethodTypeEnum {
  INTERAC_E_TRANSFER = 'InteracEtransfer',
  ZIPPY = 'Zippy',
  FUNDINGSOURCE = 'FundingSource',
}
