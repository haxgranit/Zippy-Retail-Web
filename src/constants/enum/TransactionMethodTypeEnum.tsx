// eslint-disable-next-line import/prefer-default-export
export enum TransactionMethodTypeEnum {
  ZIPPY_CASH = 'zippy-cash',
  INTERAC_E_TRANSFER = 'interac-e-transfer',
}
