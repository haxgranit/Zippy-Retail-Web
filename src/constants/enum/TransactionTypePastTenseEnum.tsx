// eslint-disable-next-line import/prefer-default-export
export enum TransactionTypePastTenseEnum {
  ALL = 'all',
  SENT = 'sent',
  RECEIVED = 'received',
  REQUESTED = 'requested',
}
