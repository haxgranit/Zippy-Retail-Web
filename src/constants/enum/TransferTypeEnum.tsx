// eslint-disable-next-line import/prefer-default-export
export enum TransferTypeEnum {
  ALL = 'all',
  SEND = 'send',
  RECEIVE = 'receive',
  REQUEST = 'request',
}
