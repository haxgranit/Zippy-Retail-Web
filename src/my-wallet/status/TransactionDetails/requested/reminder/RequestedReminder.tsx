import
{
  Button,
  Col,
  Row,
  Stack,
} from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import NumberFormat from 'react-number-format';
import { TransactionProps } from '../../TransactionStatus';

function RequestedReminder({
  transaction,
}: TransactionProps) {
  const navigate = useNavigate();

  return (
    <>
      <div className="title">
        <h3>Reminder Sent</h3>
      </div>
      <Row>
        <Col xs={6}>To</Col>
        <Col xs={6}>
          <strong>{transaction ? `${transaction.contact.firstName || ''} ${transaction.contact.lastName || ''}` : ''}</strong>
          <div>{transaction && ` (${transaction.contact.email})`}</div>
        </Col>
      </Row>
      <Row>
        <Col xs={6}>Transfer Amount</Col>
        <Col xs={6}>
          <strong className="amount">
            {transaction && transaction.amount
            && (
              <NumberFormat
                value={transaction.amount}
                defaultValue={0}
                displayType="text"
                prefix="$ "
                suffix=" CAD"
                thousandSeparator
                decimalScale={2}
                fixedDecimalScale
              />
            )}
          </strong>
        </Col>
      </Row>
      <Stack gap={3} direction="horizontal">
        <Button
          className="zippy-btn zippy-flat d-flex center w-full"
          onClick={() => navigate('/my-wallet/status')}
        >
          Back to Status
        </Button>
        <Button
          className="zippy-btn d-flex ms-auto center w-full"
          onClick={() => navigate('/my-wallet/zippy-money/request/transaction-start')}
        >
          Send Another
        </Button>
      </Stack>
    </>
  );
}

export default RequestedReminder;
