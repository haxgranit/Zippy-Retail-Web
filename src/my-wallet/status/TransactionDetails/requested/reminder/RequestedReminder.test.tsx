import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RequestedReminder from './RequestedReminder';
import { TransactionTypePastTenseEnum } from '../../../../../constants/enum/TransactionTypePastTenseEnum';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('RequestedReminder Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <RequestedReminder
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
          type={TransactionTypePastTenseEnum.ALL}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render RequestedReminder', () => {
    render(
      <BrowserRouter>
        <RequestedReminder
          transaction={undefined}
          user={undefined}
          setCurrentStatus={undefined}
          type={TransactionTypePastTenseEnum.ALL}
        />
      </BrowserRouter>,
    );

    expect(screen.getByText('To')).toBeInTheDocument();
    expect(screen.getByText('Transfer Amount')).toBeInTheDocument();
  });
});
