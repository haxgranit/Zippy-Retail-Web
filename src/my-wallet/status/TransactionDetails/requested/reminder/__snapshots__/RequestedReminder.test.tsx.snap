// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestedReminder Component matches the snapshot 1`] = `
Array [
  <div
    className="title"
  >
    <h3>
      Reminder Sent
    </h3>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      To
    </div>
    <div
      className="col-6"
    >
      <strong>
        
      </strong>
      <div />
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      Transfer Amount
    </div>
    <div
      className="col-6"
    >
      <strong
        className="amount"
      />
    </div>
  </div>,
  <div
    className="hstack gap-3"
  >
    <button
      className="zippy-btn zippy-flat d-flex center w-full btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Back to Status
    </button>
    <button
      className="zippy-btn d-flex ms-auto center w-full btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Send Another
    </button>
  </div>,
]
`;
